include/RED4ext/NativeTypes-inl.hpp

```diff
@@ -348,6 +348,39 @@ RED4EXT_INLINE void RED4ext::CurveData<T>::SetPoint(uint32_t aIndex, float aPoin
     values[aIndex] = acValue;
 }
 
+template<typename T>
+RED4EXT_INLINE void RED4ext::CurveData<T>::Resize(uint32_t aPoints) noexcept
```

Comment:
You can do it without temp storage. When you resize buffer, it copies current data.
If new size is _smaller_ than current, then copy values over `0x10 + newSize * sizeof(float)` _before_ calling `Resize()`.
If new size is _greater_ than current, then copy values over `0x10 + newSize * sizeof(float)` _after_ calling `Resize()`.